src/Jobs/IndexJob.php

```diff
@@ -13,11 +14,11 @@ class IndexJob
     protected int $id;
     protected array $values;
 
-    public function __construct(string $index, int $id, array $values)
+    public function __construct(string $index, int $id, $values)
     {
         $this->index = $index;
         $this->id = $id;
-        $this->values = $values;
+        $this->values = ($values instanceof Model ? $values->toArray() : (array)$values);
```

Comment:
```suggestion
        $this->values = ($values instanceof Arrayable ? $values->toArray() : (array)$values);
```
May be better since it explicitly ensures toArray exists without checking if it is that specific model type.